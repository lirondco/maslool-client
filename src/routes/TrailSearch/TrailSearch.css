.TrailSearch {
    margin: 2%;
    padding: 2%;
}

.TrailSearch h2 {
    margin-bottom: 1%;
}

hr {
    border: 1px solid;
}

