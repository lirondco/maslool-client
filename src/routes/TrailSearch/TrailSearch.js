import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import SearchResult from '../../components/SearchResult/SearchResult'
import TrailSearchForm from '../../components/TrailSearchForm/TrailSearchForm'
import AllTrailsContext from '../../contexts/AllTrailsContext'
import './TrailSearch.css'

export default class TrailSearch extends Component {
    state = { 
        searchActive: true
    }

    static contextType = AllTrailsContext

    handleSearchClick = () => {
        this.setState({ searchActive: false })
    }

    handleBackClick = () => {
        this.setState({ searchActive: true })
    }

    renderResults = () => {
        const { searchResult } = this.context
        return (
        <>
        <Link to='#' onClick={this.handleBackClick}>
            <h3>{'<'} Back to Search</h3>
        </Link>
        {searchResult.map(trail => {
            return (
                <SearchResult 
                    key={trail.id}
                    id={trail.id}
                    name={trail.name}
                    rating={trail.rating}
                    difficulty={trail.difficulty}
                    address_line={trail.location.address_line}
                    city={trail.location.city}
                    region={trail.location.region}
                    postal_code={trail.location.postal_code}
                />
            )
        })}
        </>
        )
    }

    render() {
        return (
            <section className='TrailSearch'>
                <h2>Base Camp</h2>
                <hr />
                {this.state.searchActive === true
                    ? <TrailSearchForm 
                    onSearchSuccess={this.handleSearchClick}
                    />
                    : this.renderResults()
                }
            </section>
        )
    }
}