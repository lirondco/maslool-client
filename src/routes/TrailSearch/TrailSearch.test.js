import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import TrailSearch from "./TrailSearch";
import { withContext } from "shallow-with-context";

describe(`TrailSearch route component`, () => {
  const context = {
    allTrails: [
      { id: 1, name: "test trail", location: { region: "Pennsylvania" } }
    ],
    searchResult: [
        { id: 1, name: "test trail", location: { region: "Pennsylvania" } },
        { id: 2, name: "other test trail", location: { region: "Iowa" } }
      ],
    clearError: () => {},
    setError: () => {},
    clearSearchResult: () => {},
  };
  const ComponentWithContext = withContext(TrailSearch, context);

  it("renders the complete page when search is active", () => {
    const wrapper = shallow(<ComponentWithContext />, { context });
    wrapper.setState({ searchActive: true });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the search result when search is not active", () => {
    const wrapper = shallow(<ComponentWithContext />, { context });
    wrapper.setState({ searchActive: false });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
