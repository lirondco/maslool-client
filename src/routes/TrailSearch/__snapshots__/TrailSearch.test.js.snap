// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrailSearch route component renders the complete page when search is active 1`] = `
<section
  className="TrailSearch"
>
  <h2>
    Base Camp
  </h2>
  <hr />
  <TrailSearchForm
    onSearchSuccess={[Function]}
  />
</section>
`;

exports[`TrailSearch route component renders the search result when search is not active 1`] = `
<section
  className="TrailSearch"
>
  <h2>
    Base Camp
  </h2>
  <hr />
  <Link
    onClick={[Function]}
    to="#"
  >
    <h3>
      &lt;
       Back to Search
    </h3>
  </Link>
  <ul>
    <li
      key="2"
    >
      <SearchResult
        id={2}
        key="2"
        name="other test trail"
        region="Iowa"
      />
    </li>
    <li
      key="1"
    >
      <SearchResult
        id={1}
        key="1"
        name="test trail"
        region="Pennsylvania"
      />
    </li>
  </ul>
</section>
`;
